src/main/scala/scalismo/ui/api/Views.scala

```diff
@@ -249,6 +249,116 @@ object VertexColorMeshView {
 
 }
 
+case class TetrahedralMeshView private[ui] (override protected[api] val peer: TetrahedralMeshNode) extends ObjectView {
+
+  type PeerType = TetrahedralMeshNode
+
+  def opacity: Double = peer.opacity.value
+
+  def opacity_=(o: Double): Unit = {
+    peer.opacity.value = o
+  }
+
+  def lineWidth: Int = peer.lineWidth.value
+
+  def lineWidth_=(width: Int): Unit = {
+    peer.lineWidth.value = width
+  }
+
+  def tetrahedralMesh: TetrahedralMesh3D = peer.source
+
+  def transformedTetrahedralMesh: TetrahedralMesh3D = peer.transformedSource
+}
+
+object TetrahedralMeshView {
+
+  implicit object FindInSceneTetrahedralMeshView$ extends FindInScene[TetrahedralMeshView] {
+    override def createView(s: SceneNode): Option[TetrahedralMeshView] = {
+      s match {
+        case peer: TetrahedralMeshNode => Some(TetrahedralMeshView(peer))
+        case _ => None
+      }
+    }
+  }
+
+  implicit object callbacksTetrahedralMeshView extends HandleCallback[TetrahedralMeshView] {
+
+    override def registerOnAdd[R](g: Group, f: TetrahedralMeshView => R): Unit = {
+      g.peer.listenTo(g.peer.triangleMeshes)
```

Comment:
Is this g.peer.triangleMesh or tetrahedralMesh? If it is triangleMesh as a generic mesh term, we could rename it to meshes

Reply:
You are right. A bug seems to have snug in. It should have been g.peer.tetrahedralMeshes.